WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/SetUpTapToPayTryPaymentPromptViewController.swift

```
@@ -0,0 +1,182 @@
+import SwiftUI
+
+/// This view controller is used when no reader is connected. It assists
+/// the merchant in connecting to a reader.
+///
+final class SetUpTapToPayTryPaymentPromptViewController: UIHostingController<SetUpTapToPayPaymentPromptView>, PaymentSettingsFlowViewModelPresenter {
+
+    private var viewModel: SetUpTapToPayTryPaymentPromptViewModel
+
+    init?(viewModel: PaymentSettingsFlowPresentedViewModel) {
+        guard let viewModel = viewModel as? SetUpTapToPayTryPaymentPromptViewModel else {
```

Should we log a `DDLogError` here? If we happen to pass any other class that conforms to `PaymentSettingsFlowPresentedViewModel` it would fail silently.

---

Good to call it out, so thank you, but it won't fail silently: this is a failable initialiser, so it will return nil and the call site will have to deal with that as appropriate.

In this case, it's the `PaymentSettingsFlowPresentingViewController`, [which will log the error when it can't make the child view controller.](https://github.com/woocommerce/woocommerce-ios/blob/dc5e351b12c4937be0c7118d6e52cd37b75aece7/WooCommerce/Classes/ViewRelated/Dashboard/Settings/CardReadersV2/PaymentSettingsFlowPresentingViewController.swift#L46)

Adding another log here would be additional noise.